Docker-dev-build

```diff
@@ -0,0 +1,135 @@
+              autoconf \
+              build-essential \
+              ca-certificates \
+              coreutils \
+              curl \
+              environment-modules \
+	            file \
+              gfortran \
+              git \
+              openssh-server \
+              python \
+              unzip \
+              vim \
+              wget \
+           && rm -rf /var/lib/apt/lists/*
+
+# load spack environment on login
+RUN echo "source $SPACK_ROOT/share/spack/setup-env.sh" \
+           > /etc/profile.d/spack.sh
+
+RUN adduser user
+USER user
+WORKDIR /home/user
+
+# install spack
+RUN git clone https://github.com/spack/spack.git
+# default branch is develop
+RUN cd spack && git checkout $SPACK_VERSION
+
+# # show which version we use
+RUN $SPACK --version
+
+# copy our package.py into the spack tree (and also example files)
+COPY spack/package.py $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/package.py
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus
+COPY spack/test/ $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+COPY check_buildlog.py /home/user 
+
+# Install and test serial and MPI versions of ocoptus via spack
+# # serial version
+
+RUN . $SPACK_ROOT/share/spack/setup-env.sh && \
+      # create a new environment for the serial version and activate it:
+      spack env create octopus-serial && \
+      spack env activate octopus-serial && \
+      # make a dev-build direcotry for the serial version:
+      mkdir ~/dev-build-serial && cd ~/dev-build-serial && \
+
+      wget https://octopus-code.org/down.php?file=12.1/octopus-12.1.tar.gz -O octopus-12.1.tar.gz &&\
+      tar --strip-components=1 -xf octopus-12.1.tar.gz -C ./&& \
+      # install the serial version of octopus:
+      # display specs of upcoming spack installation:
+      spack spec octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
+      # run the spack installation (adding it to the environment):
+      spack dev-build octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
+      # run spack smoke tests for octopus. We get an error if any of the fails:
+      spack test run --alias test_serial octopus && \
+      # display output from smoke tests (just for information):
+      spack test results -l test_serial && \
+      # show which octopus version we use (for convenience):
+      spack load octopus && octopus --version && \
+      # Run make checks
+      cd ~/dev-build-serial ; \
+      echo "-+-+-+-+ Running make check-short" ; \
+      make check-short | tee serial_check-short.log || /bin/true ; \
+      echo "-+-+-+-+ Running make check-long" ; \
+      make check-long  | tee serial_check-long.log || /bin/true ; \
+      echo "-+-+-+-+ Parsing serial_check-short.log" ; \
```

Comment:
Are the tests in `check-short` also in `check-long`? If so, we only need to run `check-long`.

Reply:
no `check-short` and `check-long` are mutually exclusive. The command `make check` is a superset which includes `check-short` and `check-long`